internal/runtime/monitoring.go

```diff
@@ -19,17 +15,6 @@ func NewServiceMonitor(serviceMonitorName, namespace string) *monitoringv1.Servi
 			Name:      serviceMonitorName,
 			Namespace: namespace,
 		},
+		Spec: monitoringv1.ServiceMonitorSpec{},
```

Comment:
I think the surrounding `NewServiceMonitor` function can be inlined into the function of the same name in `collector/service_monitor.go`.

Reply:
This separation is more trying to establish "good" practice of... a factory to create or stub a SM for any component and one specifically for collection.  I can't anticipate an alternate need at the moment but I was really trying to keep this specific separation of concern

Reply:
I can see the point of keeping this convention, although I try to avoid introducing generalizations until I need them. Some of the "stub creators" are now only used by the delete-path anymore (not this one though), I think this is what originally tripped me up in the `runtime` package.